.github/workflows/performance_harness_run.yaml

```diff
@@ -12,10 +25,131 @@ defaults:
+
+          if [[ "${{inputs.override-test-params}}" != "" ]]; then
+            echo test-params=${{inputs.override-test-params}} >> $GITHUB_OUTPUT
+          fi
+          if [[ "${{inputs.override-leap-ref}}" != "" ]]; then
+            echo leap-ref=${{inputs.override-leap-ref}} >> $GITHUB_OUTPUT
+          fi
+
+  d:
+    name: Discover Platforms
+    runs-on: ubuntu-latest
+    outputs:
+      missing-platforms: ${{steps.discover.outputs.missing-platforms}}
+      p: ${{steps.discover.outputs.platforms}}
+    steps:
+      - name: Discover Platforms
+        id: discover
+        uses: AntelopeIO/discover-platforms-action@v1
+        with:
+          platform-file: .cicd/platforms.json
+          password: ${{secrets.GITHUB_TOKEN}}
+          package-name: builders
+
+  build-platforms:
+    name: Build Platforms
+    needs: d
+    if: needs.d.outputs.missing-platforms != '[]'
+    strategy:
+      fail-fast: false
+      matrix:
+        platform: ${{fromJSON(needs.d.outputs.missing-platforms)}}
+    runs-on: ["self-hosted", "enf-x86-beefy"]
+    permissions:
+      packages: write
+      contents: read
+    steps:
+      - name: Login to Container Registry
+        uses: docker/login-action@v2
+        with:
+          registry: ghcr.io
+          username: ${{github.repository_owner}}
+          password: ${{secrets.GITHUB_TOKEN}}
+      - name: Build and push
+        uses: docker/build-push-action@v3
+        with:
+          push: true
+          tags: ${{fromJSON(needs.d.outputs.p)[matrix.platform].image}}
+          file: ${{fromJSON(needs.d.outputs.p)[matrix.platform].dockerfile}}
+
+  Build:
+    needs: [v, d, build-platforms]
+    if: always() && needs.v.result == 'success' && needs.d.result == 'success' && (needs.build-platforms.result == 'success' || needs.build-platforms.result == 'skipped')
+    runs-on: ["self-hosted", "enf-x86-beefy"]
+    container: ${{fromJSON(needs.d.outputs.p)[github.event.inputs.platform-choice].image}}
+    steps:
+        - name: Checkout leap ref
+          id: checkout
+          uses: actions/checkout@v3
+          with:
+            ref: '${{needs.v.outputs.leap-ref}}'
+            submodules: recursive
+        - name: Build
+          id: build
+          run: |
+            # https://github.com/actions/runner/issues/2033
+            chown -R $(id -u):$(id -g) $PWD
+            cmake -B build -DCMAKE_BUILD_TYPE=Release -DCMAKE_INSTALL_PREFIX=/usr -GNinja
+            cmake --build build
+            tar -pc --exclude "*.o" build | zstd --long -T0 -9 > build.tar.zst
+        - name: Upload builddir
+          uses: AntelopeIO/upload-artifact-large-chunks-action@v1
+          with:
+            name: ${{github.event.inputs.platform-choice}}-build
+            path: build.tar.zst
+
+  tests:
+    name: Tests
+    needs: [v, d, Build]
+    if: always() && needs.Build.result == 'success'
+    strategy:
+      fail-fast: false
     runs-on: ubuntu-latest
```

Comment:
This means it runs on a github owned runner, is that what you wanted?

Reply:
@spoonincode  - Correct. I was using the github owned runner for initial testing and verification of the workflow.  Now that this is closer to complete, do you have a suggestion for a runner to use (`enf-x86-beefy-long`)?  I know that most of the enf runners have time limits on them which could be problematic for performance runs depending on how a user configures the run.